File: server/models/Herb.js
const { DataTypes } = require('sequelize');
const sequelize = require('./database');

// Player herbs for alchemy
const Herb = sequelize.define('Herb', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'Users',
      key: 'id'
    }
  },
  herbId: {
    type: DataTypes.STRING,
    allowNull: false
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  count: {
    type: DataTypes.INTEGER,
    defaultValue: 1
  }
});

module.exports = Herb;

File: server-go/internal/models/herb.go
package models

// Herb 对应 Sequelize Herb 模型

type Herb struct {
	ID     uint `gorm:"primaryKey;column:id"`
	UserID uint `gorm:"column:user_id"`

	HerbID  string `gorm:"column:herb_id"`
	Name    string `gorm:"column:name"`
	Count   int    `gorm:"column:count"`
	Quality string `gorm:"column:quality"` // 品质: common, uncommon, rare, epic, legendary
}

func (Herb) TableName() string {
	return "herbs"
}
